opensearch-operator/pkg/builders/cluster.go

```diff
@@ -476,16 +699,52 @@ func NewSecurityconfigUpdateJob(instance *opsterv1.OpenSearchCluster, jobName st
 				Spec: corev1.PodSpec{
 					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
 					Containers: []corev1.Container{{
-						Name:         "updater",
-						Image:        DockerImageForCluster(instance),
-						Command:      []string{"/bin/bash", "-c"},
-						Args:         []string{arg},
-						VolumeMounts: volumeMounts,
+						Name:            "updater",
+						Image:           image.GetImage(),
+						ImagePullPolicy: image.GetImagePullPolicy(),
+						Command:         []string{"/bin/bash", "-c"},
+						Args:            []string{arg},
+						VolumeMounts:    volumeMounts,
 					}},
-					Volumes:       volumes,
-					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes:          volumes,
+					RestartPolicy:    corev1.RestartPolicyNever,
+					ImagePullSecrets: image.ImagePullSecrets,
 				},
 			},
 		},
 	}
 }
+
+func AllMastersReady(ctx context.Context, k8sClient client.Client, cr *opsterv1.OpenSearchCluster) bool {
```

Comment:
Nitpicking: Is this function and `DataNodesCount` below not rather a helper than a builder?

Reply:
It can't go in helpers because it uses a method from builders and that would cause a cyclic dependency.